### ISSUE TITLE
Feat: Version 1.0

### ISSUE THREAD

**Issue by archemist-hong** [archemist-hong/pytorch-hydra-template#2]
## Background
- 다양한 딥러닝 프로젝트에 쉽게 응용해서 사용할 수 있고, 쉽게 코드 추가/수정이 가능한  나만의 파이토치 템플릿을 만들자.

## ToDo
- [ ] 모델구조, 데이터셋, 데이터 로더 등 사용자가 쉽고 빠르게 변경할 수 있어야 함
- [ ] 외부 라이브러리와 연결하여, 여러가지 augmentation기능을 지원하고, custom augmentation도 쉽게 적용가능해야함 (+테스트 기능, 시각화)
- [ ] classification task에 적용가능해야 함
- [ ] detection task에 적용가능해야 함
- [ ] segmentation task에 적용가능해야 함
- [ ] 재현 가능한 결과를 갖도록 seed를 고정해야 함
- [ ] 여러가지 CV전략이 쉽게 구현 가능해야 함 (Hold out, K-Fold, Stratified, Group 등)
- [ ] best 모델 선정이 가능해야 함 (loss, metric)
- [ ] 여러가지 앙상블 방법을 지원해야 함 (seed, epoch, K-Fold, resize 등)
- [ ] SWA, AMP 등 쉽게 적용가능해야 함
- [ ] TTA 기능을 지원해야함 (ttach)
- [ ] Pseudo Labeling 기능
- [ ] multi-task learning을 쉽게 할 수 있도록 모델 변경, loss 변경 등이 쉬워야 함
- [ ] 실험 코드가 문제없이 돌아갈 수 있는지 적은 샘플, 작은 이미지 사이즈 등으로 몇 에폭 돌려볼 수 있는 test 기능이 필요함
- [ ] 실험 조건을 바꿔가며 여러가지 실험이 쉬지않고 연속적으로 돌아갈 수 있어야함
- [ ] down convolution
- [ ] Sliding window
- [ ] Gradient Accumulation
- [ ] 여러가지 Loss 함수
- [ ] 여러가지 Model
- [ ] Threshold searching (binary classification인 경우)
- [ ] output 이미지 확인 기능
- [ ] COCO format, Pascal VOC format 등 데이터 셋의 포맷간 변경이 쉬워야 함
- [ ] label smoothing 
- [ ] label preprocessing
- [ ] 추론 시간, 학습 시간 계산
- [ ] model summary, hydra를 활용한 실험 config 관리
- [ ] config로 부터 실험 시작
- [ ] resume기능
- [ ] logging 기능 (wandb, tensorboard, mlflow 등)
- [ ] hyper parameter searching 기능 (ray, Optuna)
